demos/https/iot_demo_https_s3_download_async.c

```diff
@@ -412,9 +412,12 @@ static void _freeRequestIndex( int i )
 /*-----------------------------------------------------------*/
 
 /**
- * @brief Free all requests in the pool that have been marked as scheduled.
+ * @brief Free all requests in the pool that have been marked as scheduled, except
+ * for the exception index.
```

Comment:
"Free all requests, with the possible exception of one of them," seems like unusual logic. Could this comment further expand on what the exceptional cases are?

Reply:
```
 * This function is used during reconnection to free any possible requests that 
 * might still have been scheduled to the HTTPS Client library to run but have 
 * not been sent yet. Reconnection happens during the _readReadyCallback(). 
 * The response's memory cannot be used until the _responseCompleteCallback() return, 
 * so during reconnection, we need to indicate that all memory except the current 
 * response is to be marked as free. 
```